### PR TITLE
A fix for tags not mimicking AWS

### PR DESCRIPTION
**Pull request by TBurnip**
This is a very simple fix for the tags problem metioned in #2148 be aware a similar problem may occur in other places. It is also not easy to find what it should act like aside from probing boto3 this means a similar problem will no doubt have to be fixed at somepoint in the future.

The fix is increadibly simple all it does is remove tagSet from the response when there are no tags to be contained within it. This means it acts the same as AWS and by virtue boto3.

### CODE REVIEW

File: moto/ec2/responses/instances.py [hunk 1]
```diff
@@ -450,6 +450,7 @@ def _security_grp_instance_attribute_handler(self):
                     </blockDeviceMapping>
                     <virtualizationType>{{ instance.virtualization_type }}</virtualizationType>
                     <clientToken>ABCDE1234567890123</clientToken>
+                    {% if instance.get_tags() != [] %}
```
Comment (mikegrima) [inline moto/ec2/responses/instances.py]: Please make this `if not instance.get_tags()` instead.

Comment (TBurnip) [reply moto/ec2/responses/instances.py]: I have updated the repository to fix this.

Comment (mikegrima) [reply moto/ec2/responses/instances.py]: Sorry! I'm a moron! It should be `if instance.get_tags()` and then list the tags.

Comment (TBurnip) [reply moto/ec2/responses/instances.py]: fixed it again. in all fairness it slipped passed me aswell

**Comment by mikegrima** [comment]
For some reason, the last test is failing.  Can you try rebasing off of master and seeing if that fixes it?

**Comment by TBurnip** [comment]
What do you mean rebasing?

**Comment by mikegrima** [comment]
@TBurnip rebase your branch with the current master branch. You would then need to push override your current branch to update the PR: https://help.github.com/en/articles/using-git-rebase-on-the-command-line.

**Comment by TBurnip** [comment]
I have now rebased it and it seems to be passing the checks.

**Pull request merged**